integration_tests/web/test_admin_conversations_restrictAccess.py

```diff
@@ -0,0 +1,91 @@
+import asyncio
+import logging
+import os
+import time
+import unittest
+
+from integration_tests.env_variable_names import SLACK_SDK_TEST_GRID_WORKSPACE_ADMIN_USER_TOKEN, \
+    SLACK_SDK_TEST_GRID_ORG_ADMIN_USER_TOKEN, \
+    SLACK_SDK_TEST_GRID_IDP_USERGROUP_ID, SLACK_SDK_TEST_GRID_TEAM_ID
+from integration_tests.helpers import async_test
+from slack import WebClient
+
+
+class TestWebClient(unittest.TestCase):
+    """Runs integration tests with real Slack API"""
+
+    def setUp(self):
+        self.logger = logging.getLogger(__name__)
+        self.org_admin_token = os.environ[SLACK_SDK_TEST_GRID_ORG_ADMIN_USER_TOKEN]
+        self.sync_client: WebClient = WebClient(
+            token=self.org_admin_token,
+            run_async=False,
+            loop=asyncio.new_event_loop()
+        )
+        self.async_client: WebClient = WebClient(token=self.org_admin_token, run_async=True)
+
+        self.team_id = os.environ[SLACK_SDK_TEST_GRID_TEAM_ID]
+        self.idp_group_id = os.environ[SLACK_SDK_TEST_GRID_IDP_USERGROUP_ID]
+
+        if not hasattr(self, "channel_id"):
```

Comment:
@kian2attari 
Here are some suggested changes. I've verified my tests work. The changes to `slack/web/base_client.py` and `slack/web/client.py` were done by `python setup.py validate`.

```diff
diff --git a/integration_tests/web/test_admin_conversations_restrictAccess.py b/integration_tests/web/test_admin_conversations_restrictAccess.py
index e5ccf46..fe469f0 100644
--- a/integration_tests/web/test_admin_conversations_restrictAccess.py
+++ b/integration_tests/web/test_admin_conversations_restrictAccess.py
@@ -29,10 +29,17 @@ class TestWebClient(unittest.TestCase):
 
         if not hasattr(self, "channel_id"):
             team_admin_token = os.environ[SLACK_SDK_TEST_GRID_WORKSPACE_ADMIN_USER_TOKEN]
-            client = WebClient(token=team_admin_token, run_async=False, loop=asyncio.new_event_loop())
+            client = WebClient(token=team_admin_token)
             # Only fetching private channels since admin.conversations.restrictAccess methods do not work for public channels
             convs = client.conversations_list(exclude_archived=True, limit=100, types="private_channel")
-            self.channel_id = next((c["id"] for c in convs["channels"] if c["name"] == "general"), None)
+            self.channel_id = next((c["id"] for c in convs["channels"] if c["name"] != "general"), None)
+            if self.channel_id is None:
+                millis = int(round(time.time() * 1000))
+                channel_name = f"private-test-channel-{millis}"
+                self.channel_id = client.conversations_create(
+                    name=channel_name,
+                    is_private=True,
+                )["channel"]["id"]
 
     def tearDown(self):
         pass
@@ -76,12 +83,16 @@ class TestWebClient(unittest.TestCase):
             team_id=self.team_id
         )
         self.assertIsNotNone(add_group)
+        # To avoid rate limiting errors
+        await asyncio.sleep(10)
 
         list_groups = await client.admin_conversations_restrictAccess_listGroups(
             team_id=self.team_id,
             channel_id=self.channel_id
         )
         self.assertIsNotNone(list_groups)
+        # To avoid rate limiting errors
+        await asyncio.sleep(10)
 
         remove_group = await client.admin_conversations_restrictAccess_removeGroup(
             channel_id=self.channel_id,
@@ -89,3 +100,5 @@ class TestWebClient(unittest.TestCase):
             team_id=self.team_id
         )
         self.assertIsNotNone(remove_group)
+        # To avoid rate limiting errors
+        await asyncio.sleep(20)
diff --git a/slack/web/base_client.py b/slack/web/base_client.py
index 3636162..ce075e1 100644
--- a/slack/web/base_client.py
+++ b/slack/web/base_client.py
@@ -686,7 +686,13 @@ class BaseClient:
 
 
 # https://api.slack.com/changelog/2020-01-deprecating-antecedents-to-the-conversations-api
-deprecated_method_prefixes_2020_01 = ["channels.", "groups.", "im.", "mpim.", "admin.conversations.whitelist."]
+deprecated_method_prefixes_2020_01 = [
+    "channels.",
+    "groups.",
+    "im.",
+    "mpim.",
+    "admin.conversations.whitelist.",
+]
 
 
 def show_2020_01_deprecation(method_name: str):
diff --git a/slack/web/client.py b/slack/web/client.py
index 4b5954c..1c3397c 100644
--- a/slack/web/client.py
+++ b/slack/web/client.py
@@ -109,17 +109,35 @@ class WebClient(BaseClient):
             "admin.apps.restricted.list", http_verb="GET", params=kwargs
         )
 
-    def admin_conversations_restrictAccess_addGroup(self, **kwargs) -> Union[Future, SlackResponse]:
+    def admin_conversations_restrictAccess_addGroup(
+        self, **kwargs
+    ) -> Union[Future, SlackResponse]:
         """Add an allowlist of IDP groups for accessing a channel."""
-        return self.api_call("admin.conversations.restrictAccess.addGroup", http_verb="GET", params=kwargs)
+        return self.api_call(
+            "admin.conversations.restrictAccess.addGroup",
+            http_verb="GET",
+            params=kwargs,
+        )
 
-    def admin_conversations_restrictAccess_listGroups(self, **kwargs) -> Union[Future, SlackResponse]:
+    def admin_conversations_restrictAccess_listGroups(
+        self, **kwargs
+    ) -> Union[Future, SlackResponse]:
         """List all IDP Groups linked to a channel."""
-        return self.api_call("admin.conversations.restrictAccess.listGroups", http_verb="GET", params=kwargs)
+        return self.api_call(
+            "admin.conversations.restrictAccess.listGroups",
+            http_verb="GET",
+            params=kwargs,
+        )
 
-    def admin_conversations_restrictAccess_removeGroup(self, **kwargs) -> Union[Future, SlackResponse]:
+    def admin_conversations_restrictAccess_removeGroup(
+        self, **kwargs
+    ) -> Union[Future, SlackResponse]:
         """Remove a linked IDP group linked from a private channel."""
-        return self.api_call("admin.conversations.restrictAccess.removeGroup", http_verb="GET", params=kwargs)
+        return self.api_call(
+            "admin.conversations.restrictAccess.removeGroup",
+            http_verb="GET",
+            params=kwargs,
+        )
 
     def admin_conversations_setTeams(self, **kwargs) -> Union[Future, SlackResponse]:
         """Set the workspaces in an Enterprise grid org that connect to a channel."""
```

Reply:
@kian2attari The changes in `setUp` method and sleep code in async test are not reflected yet